src/core/hle/service/http_c.cpp

```
@@ -2,70 +2,337 @@
-    {0x00150080, nullptr, "SetPostDataType"},
-    {0x001600C4, nullptr, "SendPostDataAscii"},
-    {0x00170144, nullptr, "SendPostDataAsciiTimeout"},
-    {0x001800C4, nullptr, "SendPostDataBinary"},
-    {0x00190144, nullptr, "SendPostDataBinaryTimeout"},
-    {0x001A0082, nullptr, "SendPostDataRaw"},
-    {0x001B0102, nullptr, "SendPOSTDataRawTimeout"},
-    {0x001C0080, nullptr, "SetPostDataEncoding"},
-    {0x001D0040, nullptr, "NotifyFinishSendPostData"},
-    {0x001E00C4, nullptr, "GetResponseHeader"},
-    {0x001F0144, nullptr, "GetResponseHeaderTimeout"},
-    {0x00200082, nullptr, "GetResponseData"},
-    {0x00210102, nullptr, "GetResponseDataTimeout"},
-    {0x00220040, nullptr, "GetResponseStatusCode"},
-    {0x002300C0, nullptr, "GetResponseStatusCodeTimeout"},
-    {0x00240082, nullptr, "AddTrustedRootCA"},
-    {0x00250080, nullptr, "AddDefaultCert"},
-    {0x00260080, nullptr, "SelectRootCertChain"},
-    {0x002700C4, nullptr, "SetClientCert"},
-    {0x002B0080, nullptr, "SetSSLOpt"},
-    {0x002C0080, nullptr, "SetSSLClearOpt"},
-    {0x002D0000, nullptr, "CreateRootCertChain"},
-    {0x002E0040, nullptr, "DestroyRootCertChain"},
-    {0x002F0082, nullptr, "RootCertChainAddCert"},
-    {0x00300080, nullptr, "RootCertChainAddDefaultCert"},
-    {0x00310080, nullptr, "RootCertChainRemoveCert"},
-    {0x00320084, nullptr, "OpenClientCertContext"},
-    {0x00330040, nullptr, "OpenDefaultClientCertContext"},
-    {0x00340040, nullptr, "CloseClientCertContext"},
-    {0x00350186, nullptr, "SetDefaultProxy"},
-    {0x00360000, nullptr, "ClearDNSCache"},
-    {0x00370080, nullptr, "SetKeepAlive"},
-    {0x003800C0, nullptr, "SetPostDataTypeSize"},
-    {0x00390000, nullptr, "Finalize"},
-};
-
-HTTP_C::HTTP_C() {
-    Register(FunctionTable);
+static const ResultCode ERROR_CONTEXT_ERROR(0xD8A0A066);
+static const ResultCode RESULT_DOWNLOADPENDING(0xd840a02b);
+
+bool HTTP_C::ContextExists(s32 context_id, IPC::RequestParser& rp) const {
```

I don't like this pattern, it hides the error code behind this function call

---

i dislike it too, but i want to avoid duplicating code, and don't know, how to make it better. There is many context-based functions